consensus/ethash/ethash.go

```diff
@@ -564,7 +574,12 @@ func NewFullFaker() *Ethash {
 
 // NewShared creates a full sized ethash PoW shared between all requesters running
 // in the same process.
-func NewShared() *Ethash {
+func NewShared(progpowNumber *big.Int) *Ethash {
```

Comment:
This doesn't really make sense if the progpow number can be changed. The point of the shared ethash engine is that it *can* be shared by every tester thread. The new code just races to who can allocate the fastest, but if multiple progpow blocksnumbers are used, it will blow up either way.

What you could do to fix it is to make the sharedEthash a `map[uint64]*Ethash`, then you can have multiple shared ones, one for each progpow transition period.